### ISSUE TITLE
v4.0.8 Windows - Unable to extract archive to a very long UNC path

### ISSUE THREAD

**Issue by dhananjay-gune** [rikyoz/bit7z#240]
bit7z v4.0.8, Win10, Visual Studio 2022 v17.11.2, C++17, x64

`BitArchiveReader.extractTo()` fails if the `outDir` is a very long UNC path e.g. 635 chars long.  
> BitException:  Failed to open the output file: The filename, directory name, or volume label syntax is incorrect.  errorCode = 123

The`failedFiles()` indicate that the file has a prefix `\\?\`   
I presume this could be the cause of failure as this prefix seems to be designated for local drives only.  
I suppose bit7z should apply the long-path-prefix designated for the UNC paths i.e. `\\?\UNC\` as described here:  
https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry  

> ![2024-09-03 00_27_43-Maximum Path Length Limitation - Win32 apps _ Microsoft Learn - Work - Microsoft](https://github.com/user-attachments/assets/6bf1c086-31b4-41b6-8593-3a736948b38d)

I could be wrong in my understanding about these prefixes!  

But what I have observed is that `fs::create_directories()` with a _long UNC path_ succeeds only if it has `\\?\UNC\` prefix and not `\\?\`.  
For a mapped network drive e.g. `S:\...long path...`, the regular prefix works i.e. `\\?\`

I use Win32 `shlwapi.h` -> `PathIsUNC()` to determine which prefix to apply.  
https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathisuncw 

Here is my sample code for your reference:  
```C++
FS::path applyLongPathPrefix(const FS::path& pathToCheck)
{
    if (isLongPathPrefixRequired(pathToCheck))
    {
        auto isUnc = PathIsUNC(pathToCheck.c_str());
        if (isUnc)
        {
            FS::path longUNCPath(WIN_LONG_PATH_PREFIX_UNC); // R"(\\?\UNC\)"
            size_t index = 2;
            tstring removedFirst2Slashes = pathToCheck.native().substr(index); // from "\\ServerName\SharedFolder" returns "ServerName\\SharedFolder"
            longUNCPath += removedFirst2Slashes;
            return longUNCPath;
        }
        FS::path longPath(WIN_LONG_PATH_PREFIX_LOCALDRIVE); // R"(\\?\)"
        longPath += pathToCheck;
        return longPath;
    }
    return pathToCheck;
}
```

Thanks for your support! 🙏

**Comment by rikyoz** [comment]
Hi!
Yeah, bit7z automatic path prefix doesn't support UNC paths.
I'll fix this in the next v4.0.9.
Thanks for pointing this out! 🙏

**Comment by rikyoz** [comment]
Fixed in [v4.0.9](https://github.com/rikyoz/bit7z/releases/tag/v4.0.9).

**Comment by dhananjay-gune** [comment]
Thanks - I will try it out in coming weeks and give you feedback
